lib/rack-cas/saml_validation_response.rb

```diff
@@ -80,13 +89,19 @@ def response
         </SOAP-ENV:Body>
       </SOAP-ENV:Envelope>~
 
-      @response = http.post(@url.request_uri, data, REQUEST_HEADERS)
+      begin
+        @response = http.post(@url.request_uri, data, REQUEST_HEADERS)
+      rescue Net::ReadTimeout
+        @response = false
```

Comment:
I would like to return something like `@response = Net::NetFailure.new('RequestDenied', "ticket '#{@ticket}' not recognized")`. 

But i didn't know how. :'(

Reply:
I don't understand this... the method calling this will expect a response object. Why returning a different object?

Reply:
I would just `raise AuthenticationFailure, "Net::ReadTimeout: ticket '#{@ticket}"`

At the end this is what we are doing at https://github.com/i4a/rack-cas/blob/1fe01a964ae3df6be07ebdd67331db5869809dc7/lib/rack-cas/saml_validation_response.rb#L20

We are introducing more complexity to the code